c++/src/capnp/layout.c++

```diff
@@ -2368,24 +2372,26 @@ MessageSizeCounts PointerReader::targetSize() const {
                             : WireHelpers::totalSize(segment, pointer, nestingLimit);
 }
 
-bool PointerReader::isNull() const {
-  return pointer == nullptr || pointer->isNull();
-}
-
-bool PointerReader::isStruct() const {
-  word* refTarget = nullptr;
-  const WirePointer* ptr = pointer;
-  SegmentReader* sgmt = segment;
-  WireHelpers::followFars(ptr, refTarget, sgmt);
-  return ptr->kind() == WirePointer::Kind::STRUCT;
-}
-
-bool PointerReader::isList() const {
-  word* refTarget = nullptr;
-  const WirePointer* ptr = pointer;
-  SegmentReader* sgmt = segment;
-  WireHelpers::followFars(ptr, refTarget, sgmt);
-  return ptr->kind() == WirePointer::Kind::LIST;
+PointerType PointerReader::getPointerType() const {
+  if(pointer->isNull()) {
```

Comment:
The previous code for isNull() also returned true if `pointer` itself was null, but you've removed that check here. Better add it back.